src/main.cpp

```diff
@@ -3069,9 +3119,18 @@ bool CheckBlock(const CBlock& block, CValidationState& state,
             return state.DoS(100, error("CheckBlock(): more than one coinbase"),
                              REJECT_INVALID, "bad-cb-multiple");
 
+    // FIXME: Need test for CheckBlock 
+    // Get branch id based on height of block which should advance tip
+    uint32_t consensusBranchId = 0;
+    int futureBlockHeight = chainActive.Height() + 1;
+    const Consensus::Params& params = Params().GetConsensus();
+    if (UPGRADE_ACTIVE == NetworkUpgradeState(futureBlockHeight, params, Consensus::UPGRADE_OVERWINTER)) {
```

Comment:
Can be simplified to `NetworkUpgradeActive(futureBlockHeight, params, Consensus::UPGRADE_OVERWINTER)`.

Can also be removed entirely; `CurrentEpochBranchId` already contains this logic internally (by virtue of Overwinter being the first upgrade), and we want to always use the branch ID of the current epoch as the consensus branch ID.